easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -328,6 +328,19 @@ def configure_step(self, srcdir=None, builddir=None):
         # ensure CMake uses EB python, not system or virtualenv python
         options.update(get_cmake_python_config_dict())
 
+        # pass the preferred host compiler, CUDA compiler, and CUDA architectures to the CUDA compiler
+        cuda_root = get_software_root('CUDA')
+        if cuda_root:
+            options['CMAKE_CUDA_HOST_COMPILER'] = which(os.getenv('CXX', 'g++'))
+            options['CMAKE_CUDA_COMPILER'] = 'nvcc'
```

Comment:
```suggestion
            options['CMAKE_CUDA_COMPILER'] = which('nvcc')
```
Googling this doesn't help much, but I have seen `CMAKE_CUDA_COMPILER` being passed as a full path in some places (and that seems consistent with what you do above)